pymedphys/_experimental/pseudonymisation/__init__.py

```diff
@@ -134,3 +138,67 @@ def is_valid_strategy_for_keywords(
         if vr not in replacement_strategy:
             return False
     return True
+
+
+def pseudonymise(dicom_input, output_path=None):
+    """Convenient API to pseudonymisation.
+    Elements whose tags are not in the pydicom dictionary will be deleted
+    PatientSex will not be modified/pseudonymised
+    For fine tune control, use anonymise_dataset() instead
+
+    Parameters
+    ----------
+    dicom_input : ``pydicom.dataset.Dataset | str | pathlib.Path``
+        Either a dataset, a path to a file or a path to a directory
+    output_path : ``str | pathlib.Path``, optional
+        If the input is a file or a path, the directory to place the
+        pseudonymised files, by default None
+
+    Returns
+    -------
+    ``pydicom.dataset.Dataset`` | ``str`` | ``list`` of ``str``
+        if the dicom_input was a dataset, return the pseudonymised dataset
+        if the dicom input was a file, return the path to the pseudonymised file.
+        if the dicom input was a directory, return the list of successfully
+        anonymised files, and return that instead of None
+    """
+    replacement_strategy = strategy.pseudonymisation_dispatch
+    identifying_keywords_for_pseudo = get_default_pseudonymisation_keywords()
+    if not is_valid_strategy_for_keywords():
+        logging.error("Pseudonymisation strategy is not valid for keywords")
+        logging.error("Please submit issue to PyMedPhys")
+        # but continue on, the data might not contain the offending keywords
+        # and if it does... there will be some kind of error raised
+    keywords_to_leave_unchanged = list("PatientSex")
+
+    if dicom_input is pydicom.dataset.Dataset:
+        pseudo_ds = anonymise_dataset(
+            dicom_input,
+            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
+            delete_unknown_tags=True,
+            replacement_strategy=replacement_strategy,
+            identifying_keywords=identifying_keywords_for_pseudo,
+        )
+        return pseudo_ds
+    else:
+        if pathlib.Path().joinpath(dicom_input).is_dir():
+            pseudonymised_file_list = anonymise_directory(
+                dicom_input,
+                output_dirpath=output_path,
+                keywords_to_leave_unchanged=keywords_to_leave_unchanged,
+                delete_unknown_tags=True,
+                replacement_strategy=replacement_strategy,
+                identifying_keywords=identifying_keywords_for_pseudo,
+            )
+            return pseudonymised_file_list
+        elif pathlib.Path().joinpath(dicom_input).is_file():
```

Comment:
maybe this should be an
else
  assert( path.is_file)

and not have a return None

basically, an assertion error if the input isn't "usable".